specs/badge.md

```diff
@@ -0,0 +1,78 @@
+# fast-badge
+
+## Overview
+
+*Badge* component is used to highlight an item and attract attention or flag status, such as “New”, “Sale”, or other short phrases or a number to represent unread messages, updates available, etc..
+
+### Use Cases
+
+Typical use cases include, but are not limited to, denoting a sale or new item, flagging an item as part of a category or representing a value of unread messages.
+  
+### Features
+
+A badge should allow the following properties:
+- `fill`, a string that passes the color value for the backplate. Defaults to "transparent".
+- `size`, the size the badge is rendered.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/badge)
+- [Material UI](https://material-ui.com/components/badges/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/badges/)
+- [Ant Design Badge](https://ant.design/components/badge/)
+- [Ant Design Lozenge](https://atlaskit.atlassian.com/packages/core/lozenge)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/badge)
+
+---
+
+### API
+
+*Component name:*
+- `fast-badge`
+
+*Attributes:*
+- `fill`: string
+- `size`: string
+
+*Slots:*
+- `default`
+
+### Anatomy and Appearance
+
+*Template:*
+```
+<slot></slot>
+```
+
+## Implementation
+
+```
+fast-badge {
+   --fill-primary: rgba(255, 0, 0, 1);
+   --fill-secondary: #00FF00;
+   --fill-transparent: transparent;
+   --size-small: 12px;
+   --size-medium: 16px;
+   --size-large: 20px;
```

Comment:
Do these set the text, width, height, etc? How does the overall badge change based on these values?

Reply:
Yeah, @nicholasrice chatted about this. Using users defined variables makes less sense for size. Since we would need to change multiple properties as you mentioned above (text, width, height). We should either go back to an `enum` for small, medium or large or have a better story for updating density for individual components (chatted with @bheston earlier about this) and decoupling sizing specific `attr` from any of our web components. I send out a PR for the badge web component to work through these very issues and currently using the `enum` approach.

Reply:
I'm not sure density would work to solve the size problem here. I may need a large badge in something like a "hero" or marketing carousel, but a smaller badge in a product card or something like that. The badge size there is relative to the control size in that scenario, but I would expect the density to stay the same on the page. I wouldn't want to change the density just around the badge implementation to get a larger badge in that scenario.

Reply:
I agree here I just wanted to mention it because it was something @bheston was thinking about.